sdk/eventhub/event-hubs/test/node/disconnects.spec.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import chai from "chai";
+const should = chai.should();
+import chaiAsPromised from "chai-as-promised";
+chai.use(chaiAsPromised);
+import { EnvVarKeys, getEnvVars } from "../utils/testUtils";
+import { EventHubClient } from "../../src/impl/eventHubClient";
+import { EventHubProducerClient } from "../../src/index";
+const env = getEnvVars();
+
+describe("disconnected", function() {
+  let client: EventHubClient;
+  const service = {
+    connectionString: env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
+    path: env[EnvVarKeys.EVENTHUB_NAME]
+  };
+  before("validate environment", function(): void {
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
+      "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_NAME],
+      "define EVENTHUB_NAME in your environment before running integration tests."
+    );
+  });
+
+  afterEach("close the connection", async function(): Promise<void> {
+    if (client) {
+      await client.close();
+    }
+  });
+
+  describe("HubRuntime", function() {
+    it("operations work after disconnect", async () => {
+      client = new EventHubClient(service.connectionString, service.path);
+      const clientConnectionContext = client["_context"];
+
+      await client.getPartitionIds({});
+      const originalConnectionId = clientConnectionContext.connectionId;
+
+      // Trigger a disconnect on the underlying connection.
+      clientConnectionContext.connection["_connection"].idle();
+
+      await client.getPartitionIds({});
```

Comment:
Might still be worth validating the results that come back, but really it's kind of a "fails completely or works" kind of call.

Reply:
I added the extra verification.